tests/components/lcn/test_cover.py

```diff
@@ -0,0 +1,390 @@
+"""Test for the LCN cover platform."""
+from unittest.mock import patch
+
+from pypck.inputs import ModStatusOutput, ModStatusRelays
+from pypck.lcn_addr import LcnAddr
+from pypck.lcn_defs import MotorReverseTime, MotorStateModifier
+
+from homeassistant.components.cover import DOMAIN as DOMAIN_COVER
+from homeassistant.components.lcn.helpers import get_device_connection
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_STOP_COVER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+    STATE_UNAVAILABLE,
+)
+
+from .conftest import MockModuleConnection
+
+
+async def test_setup_lcn_cover(hass, entry, lcn_connection):
+    """Test the setup of cover."""
+    for entity_id in (
+        "cover.cover_outputs",
+        "cover.cover_relays",
+    ):
+        state = hass.states.get(entity_id)
+        assert state is not None
+        assert state.state == STATE_OPEN
+
+
+async def test_entity_attributes(hass, entry, lcn_connection):
+    """Test the attributes of an entity."""
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entity_outputs = entity_registry.async_get("cover.cover_outputs")
+
+    assert entity_outputs
+    assert entity_outputs.unique_id == f"{entry.entry_id}-m000007-outputs"
+    assert entity_outputs.original_name == "Cover_Outputs"
+
+    entity_relays = entity_registry.async_get("cover.cover_relays")
+
+    assert entity_relays
+    assert entity_relays.unique_id == f"{entry.entry_id}-m000007-motor1"
+    assert entity_relays.original_name == "Cover_Relays"
+
+
+@patch.object(MockModuleConnection, "control_motors_outputs")
+async def test_outputs_open(control_motors_outputs, hass, lcn_connection):
+    """Test the outputs cover opens."""
+    state = hass.states.get("cover.cover_outputs")
+    state.state = STATE_CLOSED
+
+    # command failed
+    control_motors_outputs.return_value = False
+
+    await hass.services.async_call(
```

Comment:
Side note: Nowadays we prefer that service calls raise an exception if they fail. Ie, raise an exception in the service handler. The exception argument will be shown as a message to the user in the frontend.

Reply:
Thanks for the advice! In the medium term I would like to replace the service calls with device actions anyway. Then I will take that into account.

Reply:
Device actions should always be based on services. Device automations is just an extra layer on top of the automation interface that depends on core concepts such as entity states and services and the normal automation triggers, conditions and actions should still be available.